File: implementations/python/generate.py
from cid import CIDS_DIR, EXAMPLES_DIR, compute_cid


def main() -> int:
    CIDS_DIR.mkdir(exist_ok=True)
    for example in sorted(EXAMPLES_DIR.iterdir()):
        if example.is_dir():
            continue
        content = example.read_bytes()
        cid = compute_cid(content)
        destination = CIDS_DIR / cid
        destination.write_bytes(content)
        print(f"Wrote {destination.name} from {example.name}")
    return 0


if __name__ == "__main__":
    raise SystemExit(main())

File: implementations/swift/generate.swift
import Foundation

@main
struct Generate {
    static func main() {
        let fileManager = FileManager.default
        do {
            try fileManager.createDirectory(at: cidsDir, withIntermediateDirectories: true)
            let examples = try fileManager.contentsOfDirectory(at: examplesDir, includingPropertiesForKeys: [.isDirectoryKey])
                .sorted { $0.lastPathComponent < $1.lastPathComponent }
            for example in examples {
                let resourceValues = try example.resourceValues(forKeys: [.isDirectoryKey])
                if resourceValues.isDirectory == true {
                    continue
                }
                let content = try Data(contentsOf: example)
                let cid = try computeCID(content)
                let destination = cidsDir.appendingPathComponent(cid)
                try content.write(to: destination)
                print("Wrote \(destination.lastPathComponent) from \(example.lastPathComponent)")
            }
        } catch {
            fputs("Error: \(error)\n", stderr)
            exit(1)
        }
    }
}
